### PR TITLE
Publish the oc binary after each build

### PR DESCRIPTION
**Pull request by bbguimaraes**
@jupierce, @tdawson: PTAL.

### CODE REVIEW

File: scripts/merge-and-build.sh [hunk 1]
```diff
@@ -240,6 +240,32 @@ echo "=========="
...
+	set -eux
+	rpm=/mnt/rcm-guest/puddles/RHAOS/AtomicOpenShift/$1/latest/x86_64/os
+	rpm=$rpm/Packages/atomic-openshift-clients-redistributable-$2
+	rpm=$(echo "$rpm"*)
+	tmpdir=$(mktemp -dt ocbinary.XXXXXXXXXX)
+	trap "rm -rf '$tmpdir'" EXIT INT TERM
+	mkdir -p "$tmpdir/$2/"{linux,macosx,windows}
+	rpm2cpio "$rpm" | cpio -D "$tmpdir" -idm --quiet
+	cd "$tmpdir/usr/share/atomic-openshift"
+	tar --owner 0 --group 0 -C linux/ -zc oc -f "$tmpdir/$2/linux/oc.tar.gz"
```
Comment (stevekuznetsov) [inline scripts/merge-and-build.sh]: Don't we already have `tar` and `zip` archives from the Origin release build? Or do we not get those back from Brew.

Comment (tdawson) [inline scripts/merge-and-build.sh]: The only thing we get back from brew is an rpm.  We have no access to any of it's build environment.

File: scripts/merge-and-build.sh [hunk 2]
```diff
@@ -240,6 +240,32 @@ echo "=========="
...
+	tar --owner 0 --group 0 -C macosx/ -zc oc -f "$tmpdir/$2/macosx/oc.tar.gz"
+	zip --quiet --junk-path - windows/oc.exe > "$tmpdir/$2/windows/oc.zip"
+	rsync -a "$tmpdir/$2" mirror.openshift.com:/srv/pub/openshift-v3/clients/
```
Comment (tdawson) [inline scripts/merge-and-build.sh]: That's not going to work, it needs to be

rsync -aHv --delete-after --progress --no-g --omit-dir-times --chmod=Dug=rwX -e "ssh -l jenkins_aos_cd_bot -o StrictHostKeyChecking=no" "$tmpdir/$2/" use-mirror-upload.ops.rhcloud.com:/srv/pub/openshift-v3/clients//latest/$2/
ssh -l jenkins_aos_cd_bot -o StrictHostKeychecking=no use-mirror-upload.ops.rhcloud.com /usr/local/bin/push.pub.sh openshift-v3} -v

Comment (tdawson) [inline scripts/merge-and-build.sh]: Or something close  (I'm not sure about the rsync directories because I don't understand what $2 is supposed to be.

Comment (jupierce) [inline scripts/merge-and-build.sh]: @tdawson "$2" is going to be the VERSION from tito (e.g. "v3.5.0.45" with the "v" prefix removed).

File: scripts/merge-and-build.sh (hunk 1 above)
Comment (stevekuznetsov) [inline scripts/merge-and-build.sh]: Damn

File: scripts/merge-and-build.sh (hunk 2 above)
Comment (tdawson) [inline scripts/merge-and-build.sh]: Oops, I had a typo in it.  If permissions are correct, then this should work.

```
rsync -aHv --delete-after --progress --no-g --omit-dir-times --chmod=Dug=rwX -e "ssh -l jenkins_aos_cd_bot -o StrictHostKeyChecking=no" "$tmpdir/$2/" use-mirror-upload.ops.rhcloud.com:/srv/pub/openshift-v3/clients/$2/
ssh -l jenkins_aos_cd_bot -o StrictHostKeychecking=no use-mirror-upload.ops.rhcloud.com /usr/local/bin/push.pub.sh openshift-v3 -v

```

File: scripts/merge-and-build.sh [hunk 3]
```diff
@@ -240,6 +240,32 @@ echo "=========="
...
+ssh ocp-build@rcm-guest.app.eng.bos.redhat.com \
+    sh -s "$OSE_VERSION" "${VERSION#v}" <<-'EOF'
+	set -eux
+	rpm=/mnt/rcm-guest/puddles/RHAOS/AtomicOpenShift/$1/latest/x86_64/os
+	rpm=$rpm/Packages/atomic-openshift-clients-redistributable-$2
+	rpm=$(echo "$rpm"*)
+	tmpdir=$(mktemp -dt ocbinary.XXXXXXXXXX)
+	trap "rm -rf '$tmpdir'" EXIT INT TERM
+	mkdir -p "$tmpdir/$2/"{linux,macosx,windows}
+	rpm2cpio "$rpm" | cpio -D "$tmpdir" -idm --quiet
```
Comment (jupierce) [inline scripts/merge-and-build.sh]: I'm getting invalid option on -D here when running on rcp-guest.

File: scripts/merge-and-build.sh (hunk 2 above)
Comment (bbguimaraes) [inline scripts/merge-and-build.sh]: @tdawson: I imagined more elaborate commands were need, but that's exactly why
I opened the pull request.  Changes made, should be good now.

The ssh-with-command-read-from-stdin construct is a confusing, but I could not
think of a better way to run this bunch of commands on a remote server.  I've
added a couple of variables to make it a bit clearer.

File: scripts/merge-and-build.sh (hunk 3 above)
Comment (bbguimaraes) [inline scripts/merge-and-build.sh]: I didn't know, it's a relatively new option (committed in 2010, released in
2.12, 2015) not yet present on the version we have there (2.10).  No big deal,
just added an extra `cd`.

**Comment by tdawson** [comment]
With the changes, it looks good to me.

**Pull request merged**